action/execution.go

```diff
@@ -234,14 +234,37 @@ func (ex *Execution) SanityCheck() error {
 }
 
 // ToEthTx converts action to eth-compatible tx
-func (ex *Execution) ToEthTx(_ uint32) (*types.Transaction, error) {
+func (ex *Execution) ToEthTx(evmNetworkID uint32) (*types.Transaction, error) {
 	if ex.contract == EmptyAddress {
+		if len(ex.accessList) > 0 {
+			return types.NewTx(&types.AccessListTx{
+				ChainID:    big.NewInt(int64(evmNetworkID)),
+				Nonce:      ex.Nonce(),
+				GasPrice:   ex.GasPrice(),
+				Gas:        ex.GasLimit(),
+				Value:      ex.amount,
+				Data:       ex.data,
+				AccessList: ex.accessList,
+			}), nil
+		}
 		return types.NewContractCreation(ex.Nonce(), ex.amount, ex.GasLimit(), ex.GasPrice(), ex.data), nil
 	}
 	addr, err := address.FromString(ex.contract)
 	if err != nil {
 		return nil, err
 	}
 	ethAddr := common.BytesToAddress(addr.Bytes())
+	if len(ex.accessList) > 0 {
```

Comment:
`if len(ex.accessList) > 0 {...` 
These two returns can be combined into a simplified code

Reply:
yes, let's fix that in (#3990)